pkg/lib/authn/identity/service/service.go

```diff
@@ -1081,3 +1082,7 @@ func (s *Service) listSIWECandidates(siwes []*identity.SIWE) []identity.Candidat
 func (s *Service) Normalize(typ model.LoginIDKeyType, value string) (normalized string, uniqueKey string, err error) {
 	return s.LoginID.Normalize(typ, value)
 }
+
+func (s *Service) GetByKeyAndValue(loginIDKey string, loginIDValue string) (*identity.LoginID, error) {
```

Comment:
Service should always return `*identity.Info`. So you need to convert `*identity.LoginID` into `*identity.Info` here. Pay attention to error handling. This method should return `ErrIdentityNotFound` if the value cannot be used to look up an Login ID identity.